File: web/src/redux/models/UserModel.ts
export interface UserModel {
  id:              string;
  userId:          string;
  username:        string;
  profile_pic:     null;
  status:          null;
  role_id:         string;
  last_message_id: null;
  isOnline:        boolean;
  device_id:       null;
  device_token:    null;
  platform:        null;
  last_login:      null;
  createdAt:       Date;
  updatedAt:       Date;
  user:            User;
  role:            Role;
}

export interface Role {
  id:        string;
  name:      string;
  createdAt: Date;
  updatedAt: Date;
}

export interface User {
  id:           string;
  phone_number: null | string;
  email:        string;
  status:       string;
  createdAt:    Date;
  updatedAt:    Date;
  profiles? : UserModel[]
  driver_number:    null | string
}

File: chat_app/lib/modules/home/models/user_model.dart
class UserModel {
  String? id;
  String? phoneNumber;
  String? userType;
  String? driverNumber;
  int? yardId;
  String? email;
  String? status;
  String? createdAt;
  String? updatedAt;

  UserModel({
    this.id,
    this.phoneNumber,
    this.userType,
    this.driverNumber,
    this.yardId,
    this.email,
    this.status,
    this.createdAt,
    this.updatedAt,
  });

  UserModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    phoneNumber = json['phone_number'];
    userType = json['user_type'];
    driverNumber = json['driver_number'];
    yardId = json['yard_id'];
    email = json['email'];
    status = json['status'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['phone_number'] = this.phoneNumber;
    data['user_type'] = this.userType;
    data['driver_number'] = this.driverNumber;
    data['yard_id'] = this.yardId;
    data['email'] = this.email;
    data['status'] = this.status;
    data['createdAt'] = this.createdAt;
    data['updatedAt'] = this.updatedAt;
    return data;
  }
}
